src/libraries/Microsoft.PowerFx.Core/Public/Values/InMemoryRecordValue.cs

```diff
@@ -12,6 +12,7 @@ namespace Microsoft.PowerFx.Types
     internal class InMemoryRecordValue : RecordValue
     {
         private readonly IReadOnlyDictionary<string, FormulaValue> _fields;
+        private readonly IDictionary<string, FormulaValue> _fieldsDict;
```

Comment:
>_fieldsDict [](http://example.com/codeflow?start=59&length=11)

nit: consider renaming to:
  _mutableFields

That name captures the intent (that it's about mutability))